code/datums/weakrefs.dm

```diff
@@ -2,9 +2,15 @@
 	if(istype(input) && !QDELETED(input))
 		if(istype(input, /datum/weakref))
 			return input
+
 		if(!input.weak_reference)
 			input.weak_reference = new /datum/weakref(input)
 		return input.weak_reference
+	else
```

Comment:
Could you elaborate on your justification for this change? right now, WEAKREF() always returns a weakref datum. If this is causing you issues, you should fix the callers.

Reply:
It was intended for use alongside the set_pin_data proc from code/modules/integrated_electronics/core/helpers.dm, to handle individual items in lists that may not always be weakrefs.

I can get rid of this change and do additional checks in the proc itself, but I wasn't sure if that would be too clunky.